NEP: add Spending NumPy Project Funds (NEP 48) (#18454)
numpy/numpy: doc/neps/nep-0048-spending-project-funds.rst

.. _NEP48:

=====================================
NEP 48 — Spending NumPy Project Funds
=====================================

:Author: Ralf Gommers <ralf.gommers@gmail.com>
:Author: Inessa Pawson <inessa@albuscode.org>
:Author: Stefan van der Walt <stefanv@berkeley.edu>
:Status: Draft
:Type: Informational
:Created: 2021-02-07
:Resolution:

Abstract
--------

> The NumPy project has historically never received significant **unrestricted**

Reviewer: But did it receive restricted funding, was "significant" or "unrestricted" the problem before ? Do we need to invoke history before its definition here?

Reviewer 2: > But did it receive restricted funding, 

Yes, NumPy has received restricted funding. For details, refer to the section “History and current status” below.

In case you are not familiar with the internals of non-profit and academic funding, **restricted funds** are monies set aside for a particular purpose (e.g., project, educational program, etc.). They are **permanently restricted** to that purpose and cannot be used for any other expenses.  By contrast, unrestricted funds may be used for any legal purpose appropriate to the organization.

> was "significant" or "unrestricted" the problem before ? 

The lack of **unrestricted funds** was, and still remains, the biggest challenge.

> Do we need to invoke history before its definition here?

Initially, the History and Current Status section was a part of the Motivation and Scope. Its large wordcount made us concerned that it would be taking away readers’ attention from the objectives of this document. Hence, we moved it to the end. 
Would it help to insert a sentence at the beginning of this section linking to the section below? E.g., _For the overview of the NumPy funding history and current status please refer to [NumPy funding – history and current status](url)._

funding. However, that is starting to change.  This NEP aims to provide

> guidance about spending NumPy project unrestricted funds, by formulating a set

Reviewer 2: ```suggestion
guidance about spending NumPy project unrestricted funds by formulating a set
```

of principles about *what* to pay for and *who* to pay. It will also touch on
how decisions regarding spending funds get made, how funds get administered,
and transparency around these topics.

Motivation and Scope
--------------------

> 

Reviewer 2: ```suggestion
NumPy is a fiscally sponsored project of NumFOCUS, a 501(c)(3) nonprofit organization headquartered in Austin, TX. Therefore, for all legal and accounting matters the NumPy project has to follow the rules and regulations for US nonprofits. All nonprofit donations are classified into two categories: **unrestricted funds** which may be used for any legal purpose appropriate to the organization and **restricted funds**, monies set aside for a particular purpose (e.g., project, educational program, etc.). It’s important to note that restricted funds are **permanently restricted** to that purpose and cannot be used for any other expenses.  

For the detailed timeline of NumPy funding refer to [NumPy funding – history and current status](#history-and-current-status).
```

Reviewer: This reads much much better now for the rest of the history to follow.

Author: Thanks @InessaPawson. I'm not sure the "permanently restricted" bit is technically true or necessary, probably best left out. But for now this added paragraph helps people understand the rest better, so I'll commit it as-is.

Reviewer 2: @rgommers You are right, it's better to leave the fourth sentence out.

> In its 16+ year history, the NumPy project has only spent on the order of

Reviewer 2: ### @rgommers Please edit as follows:

<strike>In its 16+ year history</strike> Since its inception and until 2020, the NumPy project has only spent on the order of $10,000 USD of funds that were not restricted to a particular program. Project income of this type has been relying on donations from individuals and, from <strike>May</strike> mid 2019, recurring monthly contributions from Tidelift.

> $10,000 USD of funds that were not restricted to a particular program.  Project

Reviewer: What is a program?

Reviewer 2: It’s another commonly used term in non-profit and academic funding. Would the word “purpose” be less confusing?

Reviewer 3: Since we expect this document to live for many years, I think rough dates would be more appropriate:
```
Since its inception and until 2020, the NumPy project has only seen about $10,000 of unrestricted* funding.

* unrestricted funding is money contributed to the project with no conditions for use: the project can decide to spend it on equipment, salary, conferences, or any other purpose as the project sees fit.
```

income of this type has been relying on donations from individuals and, from

> May 2019, recurring monthly contributions from Tidelift. By the end of 2020,

Reviewer 2: ```suggestion
mid 2019, recurring monthly contributions from Tidelift. By the end of 2020,
```

the Tidelift contributions increased to $3,000/month, and there's also a
potential for an increase of donations and grants going directly to the
project. Having a clear set of principles around how to use these funds will
facilitate spending them fairly and effectively. Additionally, it will make it
easier to solicit donations and other contributions.

A key assumption this NEP makes is that NumPy remains a largely
volunteer-driven project, and that the project funds are not enough to employ
maintainers full-time. If funding increases to the point where that assumption
is no longer true, this NEP should be updated.

In scope for this NEP are:

- Principles of spending project funds: what to pay for, and who to pay.
- Describing how NumPy's funds get administered.
- Describing how decisions to spend funds get proposed and made.

Out of scope for this NEP are:

- Making any decisions about spending project funds on a specific project or
  activity.

> - Principles for spending funds that are intended for NumPy development, but

Reviewer: I think the definition of un-unrestricted funds are in this sentence but I couldn't fully understand it. Though I feel what is mentioned here.

Reviewer 2: I've already covered it in this comment: https://github.com/numpy/numpy/pull/18454/commits/cf9a5db9bca7af3665c78c071cff49d36214481b#r580023603. 
Non-profit and academic funding is the world of its own.:)

Reviewer 3: If these terms have specific meanings that are not generally understood, they should be defined, maybe as footnotes.

Reviewer 2: @mattip @ilayn What do you think of a short glossary prefacing the document instead of the footnotes?

Reviewer 3: Any supplementary information like a glossary should be at the end, not the beginning.

Reviewer: If this is going to be read and not filed away I would expect a definition at the beginning. This is not an article after all so footnote I would really not bother with. It doesn't have to be a definition section anyways. I think the academic style is creeping in here.

Author: This is exactly why I had the history section at the start. Basically, unrestricted funds are coming from donations, Tidelift monthly payments, GSoC mentoring fees, and other such things - income for the project that goes into our account and we don't already have a donation for.

This contrasts with grants and other types of contracts, which typically say "you get $xxx,000 for deliverables Y or activities Z".

Reviewer: So then if we connect that history to what you just wrote here, I'm sold. I just need to get an understanding of what restricted means such that the rest is easy to parse.

Reviewer 2: > If this is going to be read and not filed away 

If we are successful in our fundraising efforts, I expect this NEP to be used on a regular basis.

  don't fall in the category of NumPy unrestricted funds. This includes most of
  the grant funding, which is usually earmarked for certain
  activities/deliverables and goes to an Institutional Partner rather than
  directly to the NumPy project, and companies or institutions funding specific
  features.

>   *Rationale: As a project, we have no direct control over how this work gets

Reviewer: Which work ? Is that the someone hired by someone else but hired to work on NumPy or something?

Reviewer 2: Correct.

Reviewer 2: @ilayn Some of these contributions are much like a very delicious dish at a large potluck party: you have no idea who cooked it, at whose kitchen, and who paid for the ingredients.;)

Reviewer: Yes that's something I can imagine. My point was more about there is a work involved but it doesn't read from the previous paragraph. Who defines the work, is an institutional body decides to suddenly go work on NumPy and gets access immediately? I think it needs a bit of clarification that even when funding is involved not all requests are taken in and there is some more elaborate screening happening. 

Here it reads like company A popped up, paid us X amount of money, and implemented `numpy.<insert lib name>` which is definitely not the case.

Reviewer 2: @ilayn True, it’s definitely not the case.
Since it’s out of scope, we didn’t go into details.

Author: adding a clarification "(at least formally, until issues or PRs show up)".

  executed. In some cases, we may not even know the contributions were funded
  or done by an employee on work time. (Whether that's the case or not should
  not change how we approach a contribution).  For grants though, we do expect

>   the PI and funded team to align their work with the project's needs and be

Reviewer: PI is a niche academic (and mostly US) term and does not mean anything to external fiscal bodies worldwide should they wish to examine this NEP.

Reviewer 2: @ilayn Fair point. I’ll reword it.

Reviewer 2: @rgommers Please edit as follows:
<strike>...the PI and funded team to align their work with the project's needs... </strike>
...the research/project leader and funded team to align their work with the needs of NumPy...

Principles of spending project funds
------------------------------------

NumPy will likely always be a project with many times more volunteer
contributors than funded people. Therefore having those funded people operate
in ways that attract more volunteers and enhance their participation experience
is critical. That key principle motivates many of the more detailed principles
given below for what to pay for and whom to pay.

What to pay for
```````````````

1. Pay for things that are important *and* otherwise won't get done.
   *Rationale: there is way more to be done than there are funds to do all
   those things. So count on interested volunteers or external sponsored work
   to do many of those things.*
2. Plan for sustainability. Don't rely on money always being there.
3. Consider potential positive benefits for NumPy maintainers and contributors,
   maintainers of other projects, end users, and other stakeholders like
   packagers and educators.
4. Think broadly. There's more to a project than code: websites, documentation,
   community building, governance - it's all important.
5. For proposed funded work, include paid time for others to review your work

>    if such review is expected to be significant effort - do not just increase

Reviewer 2: _...to be **a** significant effort..._

   the load on volunteer maintainers.
   *Rationale: we want the effect of spending funds to be positive for
   everyone, not just for the people getting paid. This is also a matter of
   fairness.*

When considering development work, principle (1) implies that priority should
be giving to (a) the most boring/painful tasks that no one likes doing, and to
necessary structural changes to the code base that are too large to be done by
a volunteer in a reasonable amount of time.

> There are also a large amount of tasks, activities, and projects outside of

Reviewer 4: Consider "many" instead of "a large amount".

development work that are important and could enhance the project or community

> - think for example of user surveys, translations, outreach, dedicated

Reviewer 2: ```suggestion
- think of, for example, user surveys, translations, outreach, dedicated
```

mentoring of newcomers, community organization, website improvements, and
administrative tasks.

Time of people to perform tasks is also not the only thing that funds can be
used for: expenses for in-person developer meetings or sprints, hosted hardware
for benchmarking or development work, and CI or other software services could
all be good candidates to spend funds on.

Whom to pay
```````````

1. All else being equal, give preference to existing maintainers/contributors.

> 2. Consider this an opportunity to make the project more diverse.

Reviewer 5: Although I don't read it that way, 1. and 2. could be considered to be going in different directions. Presumably, there's a point at which you save an order of magnitude of time (and money) by using an internal volunteer of course.

Reviewer 2: @tylerjereddy As you know, the number of current active NumPy volunteers/contributors is very low.
In cases where there is no potential candidate/s from the pool of existing volunteers, the Council is strongly encouraged to advertise an existing paid work opportunity with the underrepresented groups in tech.
We have already established good relationships with DataUmbrella, GeoLatinas, PyLadies, and WiMLDS.

Author: Rephrasing point 2 to: _"When looking outside of the current team, consider this an opportunity to make the project more diverse."_

3. Pay attention to the following when considering paying someone:

   - the necessary technical or domain-specific skills to execute the tasks,
   - communication and self-management skills,
   - experience contributing to and working with open source projects.

It will likely depend on the project/tasks whether there's already a clear best
candidate within the NumPy team, or whether we look for new people to get
involved. Before making any decisions, the decision makers should think about
whether an opportunity should be advertised to give a wider group of people a

> chance to throw their hat in the ring for it.

Reviewer 4: ```suggestion
chance to apply for it.
```

Compensating fairly

> ```````````````````

Author: Given the amount of discussion, let's put in a `.. note::` here that this section will not be final until we have tried these guidelines at least 2-3 times and are happy with them (and if not, we adjust).

Paying people fairly is a difficult topic. Therefore, we will only offer some
guidance here. Final decisions will always have to be considered and approved
by the group of people that bears this responsibility (according to the
current NumPy governance structure, this would be the NumPy Steering Council).

Discussions on employee compensation tend to be dominated by two narratives:
"pay local market rates" and "same work -- same pay".

We consider them both extreme:

- "Same work -- same pay" is unfair to people living in locations with a higher
  cost of living. For example, the average rent for a single family apartment

>   can differ by a large factor (from a few hundred dollar to thousands of

Reviewer 2: _...a few hundred dollar**s**..._

for guidelines on remuneration policies and compensation levels. Our findings
show that most smaller non-profits tend to pay a median salary/wage. We
recognize merit in this approach: applying candidates are likely to have a
genuine interest in open source, rather than to be motivated purely by
financial incentives.

Considering all of the above, we will use the following guidelines for
determining compensation:

1. Aim to compensate people appropriately, up to a level that's expected for
   senior engineers or other professionals as applicable.
2. Establish a compensation cap of $125,000 USD that cannot be exceeded even
   for the residents from the most expensive/competitive locations ([#f-pay]_).
3. For equivalent work and seniority,  a pay differential between locations
   should never be more than 2x.
   For example, if we pay $110,000 USD to a senior-level developer from New
   York, for equivalent work a senior-level developer from South-East Asia
   should be paid at least $55,000 USD. To compare locations, we will use
   `Numbeo Cost of Living calculator <https://www.numbeo.com/cost-of-living/>`__

>    (or its equivalent).

Reviewer 5: Sorry if it is clarified somewhere below--given two approximately equivalently-skilled candidates working in locations that might justify paying one candidate substantially less than the other, will the lower-burden candidate have preference on the basis of being able to perform x % more work for the available funds?

That is kind of like dumping the outsourcing debate on this NEP, but the the cost of living matter is indeed discussed here in some detail so it is hard to avoid thinking that this might be an ethical dilemma of sorts (it is addressed by the fully-remote i.e., GitLab policies?).

Reviewer 2: @tylerjereddy I understand your concerns. Personally, I encourage the NumPy Steering Council, the body that is currently in charge of hiring decisions, to be “location-blind” in a selection of contractors and choose the best candidate for the job/task. After all, NumPy is a not-for-profit project.

Reviewer 6: Location is one factor that determines how contractors spend their money. Factors like supporting family could be considered even more important than location as the contractors' families may vary a lot in size and nature. That said, I think the location and other similar factors should be discarded when negotiating the contract terms. To get the job done well and quickly defines the efficiency of spending project funds, not how contractors are likely to spend their pays - it is nobody's business, imho.

Reviewer 2: @tylerjereddy @pearu Thank you for taking part in this very important discussion. Would you mind copying your comments to the NumPy mailing list? According to the NumPy governance document (https://numpy.org/doc/stable/dev/governance/governance.html), this type of discussion should take place there. 
You are welcome to leave comments here, on GitHub, if you catch a typo or have minor suggestions for improving the text of this document.

Reviewer 6: https://mail.python.org/pipermail/numpy-discussion/2021-February/081531.html

Author: Given the amount of discussion and that we don't have experience with doing this yet, I added:

<img width="534" alt="image" src="https://user-images.githubusercontent.com/98330/109399473-50c83580-7943-11eb-9f08-188b9a754c71.png">

Some other considerations:

- Often, compensated work is offered for a limited amount of hours or fixed
  term. In those cases, consider compensation equivalent to a remuneration
  package that comes with permanent employment (e.g., one month of work should
  be compensated by at most 1/12th of a full-year salary + benefits).
- When comparing rates, an individual contractor should typically make 20% more
  than someone who is employed since they have to take care of their benefits
  and accounting on their own.
- Some people may be happy with one-off payments towards a particular

>   deliverable (e.g., hiring a cleaner or some other service to use the saved

Reviewer 4: It seems natural to read "hiring a cleaner" as an example "one-off payments toward a particular deliverable". That is not what is intended, though.

Author: Good point, that's a bit mixed up. I had two related but different things in mind:

1. A deliverable like "do a release" or "triage all open issues for label X", for a fixed price of $xxxx.
2. We've had cases where people motivated, for example, a NumFOCUS Small Development Grant with "I need X hours to complete proposed work Y". To free up that time, I'd like $2500 for 100 hours of time for my cleaner so I can justify working on Y.

Reviewer 2: > 2. We've had cases where people motivated, for example, a NumFOCUS Small Development Grant with "I need X hours to complete proposed work Y". To free up that time, I'd like $2500 for 100 hours of time for my cleaner so I can justify working on Y.

It’s quite creative!:) However, I’m not convinced that such a corner case should be included in this document. I’ll leave the final decision up to you, @rgommers.

Reviewer 7: If you want a concrete example to make the argument more compelling, you could mention childcare as an extra cost to be factored in.

Author: I like the childcare example too - seems good to keep.

every six months to review all the project related transactions. (In practice,
there have been so few transactions that we skipped some of these meetings.)

The existing process is time-consuming and error-prone. More transparency and
automation are desirable.

Transparency about project funds and in decision making
```````````````````````````````````````````````````````

**To discuss: do we want full transparency by publishing our accounts,
transparency to everyone on a NumPy team, or some other level?**

Ralf: I'd personally like it to be fully transparent, like through Open
Collective, so the whole community can see current balance, income and expenses
paid out at any moment in time. Moving to Open Collective is nontrivial,
however we can publish the data elsewhere for now if we'd want to.
*Note: Google Season of Docs this year requires having an Open Collective
account, so this is likely to happen soon enough.*

Stefan/Inessa: at least a summary overview should be fully public, and all
transactions should be visible to the Steering Council. Full transparency of
all transactions is probably fine, but not necessary.

> *The options here may be determined by the accounting system and amount of

Reviewer 2: _and **the** amount of effort_

effort required.*

> History and current status

Reviewer 2: @rgommers Let's rename it to “NumPy funding – history and current status”.

  Institute of Data Science. Work performed during the period 2017-2020;
  PI Stéfan van der Walt.
- Two grants from the Chan Zuckerberg Foundation to NumFOCUS, for a combined
  amount of $335k. Work performed during the period 2020-2021; PI's Ralf
  Gommers (first grant) and Melissa Mendonça (second grant).

From 2012 onwards NumPy has been a fiscally sponsored project of NumFOCUS.
Note that fiscal sponsorship doesn't mean NumPy gets funding, rather that it
can receive funds under the umbrella of a nonprofit. See `NumFOCUS Project
Support <https://numfocus.org/projects-overview>`__ for more details.

Only since 2017 has the NumPy website displayed a "Donate" button, and since
2019 the NumPy repositories have had the GitHub Sponsors button. Before that,
it was possible to donate to NumPy on the NumFOCUS website. The sum total of
donations from individuals to NumPy for 2017-2020 was about $6,100.

From May 2019 onwards, Tidelift has supported NumPy financially as part of
its "managed open source" business model. From May 2019 till July 2020 this was
$1,000/month, and it started steadily growing after that to about $3,000/month
(as of Feb 2021).

> Finally there have been other incidental project income, for example some book

Reviewer 2: _Finally **,** there **has** been other incidental project income, for example **,** some..._

royalties from Packt Publishing, GSoC mentoring fees from Google, and
merchandise sales revenue through the NumFOCUS web shop. All of these were
small (two or three figure) amounts.

This brings the total amount of project income which did not already have a
spending target to about $35,000. Most of that is recent, from Tidelift.
Over the past 1.5 years we spent about $10,000 for work on the new NumPy
website and Sphinx theme. Those spending decisions were made by the NumPy
Steering Council and announced on the mailing list.

That leaves about $25,000 in available funds at the time of writing, and

> that amount is current growing at a rate of about $3,000/month.

Reviewer 2: _...current**ly** growing..._

:Type: Informational
:Created: 2021-02-07
:Resolution:

Abstract
--------

The NumPy project has historically never received significant **unrestricted**
funding. However, that is starting to change.  This NEP aims to provide
guidance about spending NumPy project unrestricted funds by formulating a set
of principles about *what* to pay for and *who* to pay. It will also touch on
how decisions regarding spending funds get made, how funds get administered,
and transparency around these topics.

Motivation and Scope
--------------------

NumPy is a fiscally sponsored project of NumFOCUS, a 501(c)(3) nonprofit organization headquartered in Austin, TX. Therefore, for all legal and accounting matters the NumPy project has to follow the rules and regulations for US nonprofits. All nonprofit donations are classified into two categories: **unrestricted funds** which may be used for any legal purpose appropriate to the organization and **restricted funds**, monies set aside for a particular purpose (e.g., project, educational program, etc.). It’s important to note that restricted funds are **permanently restricted** to that purpose and cannot be used for any other expenses.  

> For the detailed timeline of NumPy funding refer to [NumPy funding – history and current status](#history-and-current-status).

Reviewer 2: ```suggestion
For the detailed timeline of NumPy funding refer to [NumPy funding – history and current status](#NumPy funding – history and current status).
```

mentoring of newcomers, community organization, website improvements, and
administrative tasks.

Time of people to perform tasks is also not the only thing that funds can be
used for: expenses for in-person developer meetings or sprints, hosted hardware
for benchmarking or development work, and CI or other software services could
all be good candidates to spend funds on.

Whom to pay
```````````

1. All else being equal, give preference to existing maintainers/contributors.
2. Consider this an opportunity to make the project more diverse.
3. Pay attention to the following when considering paying someone:

   - the necessary technical or domain-specific skills to execute the tasks,
   - communication and self-management skills,
   - experience contributing to and working with open source projects.

It will likely depend on the project/tasks whether there's already a clear best
candidate within the NumPy team, or whether we look for new people to get
involved. Before making any decisions, the decision makers should think about
whether an opportunity should be advertised to give a wider group of people a
chance to apply for it.

Compensating fairly
```````````````````

> 

Reviewer 2: ```suggestion
.. note::
The authors of this NEP are not fully settled on the approach described below and would like to put it in practice at least with 2-3 hires before submitting a final recommendation.
```

> Paying people fairly is a difficult topic. Therefore, we will only offer some

Reviewer 2: ```suggestion
Paying people fairly is a difficult topic, especially when it comes to distributed teams. Therefore, we will only offer some
```

guidance here. Final decisions will always have to be considered and approved
by the group of people that bears this responsibility (according to the
current NumPy governance structure, this would be the NumPy Steering Council).

> Discussions on employee compensation tend to be dominated by two narratives:

Reviewer 2: ```suggestion
Discussions on remote employee compensation tend to be dominated by two narratives:
```

   senior engineers or other professionals as applicable.
2. Establish a compensation cap of $125,000 USD that cannot be exceeded even
   for the residents from the most expensive/competitive locations ([#f-pay]_).
3. For equivalent work and seniority,  a pay differential between locations
   should never be more than 2x.
   For example, if we pay $110,000 USD to a senior-level developer from New
   York, for equivalent work a senior-level developer from South-East Asia
   should be paid at least $55,000 USD. To compare locations, we will use
   `Numbeo Cost of Living calculator <https://www.numbeo.com/cost-of-living/>`__
   (or its equivalent).

Some other considerations:

- Often, compensated work is offered for a limited amount of hours or fixed
  term. In those cases, consider compensation equivalent to a remuneration
  package that comes with permanent employment (e.g., one month of work should
  be compensated by at most 1/12th of a full-year salary + benefits).
- When comparing rates, an individual contractor should typically make 20% more
  than someone who is employed since they have to take care of their benefits
  and accounting on their own.

> - Some people may be happy with one-off payments towards a particular

Reviewer 2: ```suggestion
- Some individual contributors may be happy with one-off payments for the work on a particular deliverable (e.g., triaging all open issues for label X). This should be compensated at a lower rate compared to an individual contractor.
```

>   deliverable (e.g., hiring a cleaner or some other service to use the saved

Reviewer 2: ```suggestion

```

>   time for work on open source). This should be compensated at a lower rate

Reviewer 2: ```suggestion
```

>   compared to an individual contractor.

Reviewer 2: ```suggestion

```

it was possible to donate to NumPy on the NumFOCUS website. The sum total of
donations from individuals to NumPy for 2017-2020 was about $6,100.

From May 2019 onwards, Tidelift has supported NumPy financially as part of
its "managed open source" business model. From May 2019 till July 2020 this was
$1,000/month, and it started steadily growing after that to about $3,000/month
(as of Feb 2021).

Finally, there has been other incidental project income, for example, some book
royalties from Packt Publishing, GSoC mentoring fees from Google, and
merchandise sales revenue through the NumFOCUS web shop. All of these were
small (two or three figure) amounts.

This brings the total amount of project income which did not already have a
spending target to about $35,000. Most of that is recent, from Tidelift.
Over the past 1.5 years we spent about $10,000 for work on the new NumPy
website and Sphinx theme. Those spending decisions were made by the NumPy
Steering Council and announced on the mailing list.

That leaves about $25,000 in available funds at the time of writing, and
that amount is currently growing at a rate of about $3,000/month.

Related Work
------------

> 

Reviewer 2: For this paragraph, consider the following rewrite:

```suggestion
See references. We assume that other open source projects have also developed guidelines on spending project funds. However, we were unable to find any examples at the time of writing.
```

Author: thanks, adopted this and other suggestions.

   *Rationale: there is way more to be done than there are funds to do all
   those things. So count on interested volunteers or external sponsored work
   to do many of those things.*
2. Plan for sustainability. Don't rely on money always being there.
3. Consider potential positive benefits for NumPy maintainers and contributors,
   maintainers of other projects, end users, and other stakeholders like
   packagers and educators.
4. Think broadly. There's more to a project than code: websites, documentation,
   community building, governance - it's all important.
5. For proposed funded work, include paid time for others to review your work
   if such review is expected to be significant effort - do not just increase
   the load on volunteer maintainers.
   *Rationale: we want the effect of spending funds to be positive for
   everyone, not just for the people getting paid. This is also a matter of
   fairness.*

When considering development work, principle (1) implies that priority should
be giving to (a) the most boring/painful tasks that no one likes doing, and to
necessary structural changes to the code base that are too large to be done by

> a volunteer in a reasonable amount of time.

Author: This indeed can happen. The first principle in "what to pay for" says to pay for things that otherwise won't get done. One reason something doesn't get done is because the required skills are missing in the project/community.

I think we are missing skills, but more outside of development work. It's very well possible that it's a combination of a rare skill and no one in the team with bandwidth + motivation to address the need.

So we look outside of the community to find the right person. I think that's addressed in the next section ("who to pay").

development work that are important and could enhance the project or community
- think of, for example, user surveys, translations, outreach, dedicated
mentoring of newcomers, community organization, website improvements, and
administrative tasks.

Time of people to perform tasks is also not the only thing that funds can be
used for: expenses for in-person developer meetings or sprints, hosted hardware
for benchmarking or development work, and CI or other software services could
all be good candidates to spend funds on.

Whom to pay
```````````

1. All else being equal, give preference to existing maintainers/contributors.
2. When looking outside of the current team, consider this an opportunity to
   make the project more diverse.
3. Pay attention to the following when considering paying someone:

   - the necessary technical or domain-specific skills to execute the tasks,
   - communication and self-management skills,
   - experience contributing to and working with open source projects.

It will likely depend on the project/tasks whether there's already a clear best
candidate within the NumPy team, or whether we look for new people to get

> involved. Before making any decisions, the decision makers should think about

Author: For now the model is: discussion in public on the "what to do", and then the Steering Council decides on who and how much.

My plan is to change that slightly, but that's a future governance NEP.

> involved. Before making any decisions, the decision makers should think about

Author: adding `` (see :ref:`section-what-to-pay-for`) ``

  their time on certain tasks may turn out to be more effective. Anything that
  helps the project and community grow and improve is worth considering.
- Transparency helps. If everyone involved is comfortable sharing their
  compensation levels with the rest of the team (or better make it public),
  it's least likely to be way off the mark for fairness.

We highly recommend that the individuals involved in decision-making about
hiring and compensation peruse the content of the References section of this
NEP. It offers a lot of helpful advice on this topic.

Defining fundable activities and projects
-----------------------------------------

We'd like to have a broader set of fundable ideas that we will prioritize with
input from NumPy team members and the wider community. All ideas will be
documented on a single wiki page. Anyone may propose an idea. Only members of a
NumPy team may edit the wiki page.

Each listed idea must meet the following requirements:

1. It must be clearly scoped: its description must explain the importance to
   the project, referencing the NumPy Roadmap if possible, the items to pay for

>    or activities and deliverables, and why it should be a funded activity.

Author: adding `` (see :ref:`section-what-to-pay-for`) ``
